spec/e2e/support/crawl.js

```diff
@@ -0,0 +1,11 @@
+import { get } from 'http';
+import saxophonist from 'saxophonist';
+
+export default async function* crawl(sitemap) {
+  const response = await new Promise((resolve) => get(sitemap, resolve));
+  const locs = response.pipe(saxophonist('loc'));
+
+  for await (const loc of locs) {
+    yield loc.text;
+  }
```

Comment:
would this be simpler if we moved the `await` to the assignment?

```suggestion
  const locs = await response.pipe(saxophonist('loc'));

  for (const loc of locs) {
    yield loc.text;
  }
```

Reply:
> would this be simpler if we moved the `await` to the assignment?

I'd be curious to hear what specifically improves simplicity, though at a glance I don't think your proposal would work as expected, since a stream cannot be `await`-ed directly.

The syntax here is essentially a syntactic sugar on top of Node streams. There's a good article about it here:

https://2ality.com/2019/11/nodejs-streams-async-iteration.html

If I understand correctly, the main difference between what you're proposing and what happens here is that we can yield chunks as they're received / parsed, rather than awaiting them all and only then iterating through the set of them.

We could do that, and it probably wouldn't be terribly non-performant, but since Node's `http.get` response is a stream anyways, it's a pretty good fit to use the async iterators. Alternatively, we could use a package like [`axios`](https://www.npmjs.com/package/axios) or [`node-fetch`](https://github.com/node-fetch/node-fetch) to resolve the XML as a promise (which we'd still need to parse).

Reply:
I didn't realize iterating through the response was also a stream, TIL thanks